src/client/hud.cpp

```diff
@@ -314,12 +314,12 @@ bool Hud::hasElementOfType(HudElementType type)
 }
 
 // Calculates screen position of waypoint. Returns true if waypoint is visible (in front of the player), else false.
-bool Hud::calculateScreenPos(const v3s16 &camera_offset, HudElement *e, v2s32 *pos)
+bool Hud::calculateScreenPos(const v3pos_t &camera_offset, HudElement *e, v2s32 *pos)
 {
-	v3f w_pos = e->world_pos * BS;
+	v3opos_t w_pos_o = v3fToOpos(e->world_pos) * BS;
```

Comment:
Why is the world pos a `v3f`? Pretty sure that will break.

Reply:
for less changes. it can be fixed later.
nothing can breaks here, only precision in huge coords.
float values will align to some grid, its subject for future fixes (review and remove most v3ftoopos and back conversions)